docs/src/alghom.md

```diff
@@ -39,11 +39,11 @@ julia> L = FiniteField(3, 2, "a")
 
 julia> R, (x, y, z, w) = polynomial_ring(L[1], ["x", "y", "z", "w"];
                                     ordering=:negdegrevlex)
-(Singular Polynomial Ring (9,a),(x,y,z,w),(ds(4),C), spoly{n_GF}[x, y, z, w])
+(Singular Polynomial Ring (9,@OSCAR@a@1),(@OSCAR@x@1,@OSCAR@y@1,@OSCAR@z@1,@OSCAR@w@1),(ds(4),C), spoly{n_GF}[x, y, z, w])
```

Comment:
I don't think we should update the variable names here. Instead I would perform this kind of renaming in Oscar.jl

Reply:
Here in the docstring? Or altogether?

Reply:
I at least agree that if I am renaming the variables in Singular.jl, then prepending `@OSCAR@` is not the right approach. But then the question becomes, when do they get renamed? I went with this because Singular.jl already had a helper function to rename "bad" variable names. I can create one in Oscar.jl, but when to call it becomes an issue. Because this issue is not just with computing primary decomposition from Singular, we could potentially run into these name clashes any time we call the Singular library.

Unrelated, at the moment this renaming is causing a segfault in the tests, during one of the calls to `libSingular.algExt_GetMinpoly`, trying to figure out what is going on with that.

Reply:
Looking into this more, the renaming is sort of a fundamental part of Singular.jl. The type constructors convert to a "good" list of variable names and write them to the Singular objects that they wrap, while maintaining a vector of the corresponding symbols used for the variable names in Julia. I think it would probably break some things to remove this from Singular.jl, and would require some type piracy to handle all of this from the Oscar.jl side.

I think Hans has helped identify the source of the segfault, but regardless, I would honestly vote for just considering *every* variable name bad, and replace it by `def*<number>` where `def` is the (already implemented) default taken by `rename_symbol`. The reasons being, 
1.  it's infeasible to predict all of the possible identifiers that will create a clash in the Singular library and
2. from the Julia side, we shouldn't care what Singular is calling these variables internally.

There are only 3 times when we might see these Singular variable names, and it will always be I think an issue that should be fixed:
1. We have Singular actually printing the output for some reason.
2. We have a badly defined print routine for an object that is pulling the variable names from the wrapped Singular object instead of from the list of julia symbols the object is using.
3. We for some reason are creating a julia object directly from a pointer to a Singular object, in this case it reads the variable names from Singular.

Related to 2. above, the Singular.jl polynomial rings have a bad print routine that shows the wrapped variable names (and is in general unintelligeable):
```
julia> using Singular;
julia> F, (a,) = FunctionField(QQ,["a"]);
julia> K, a = AlgebraicExtensionField(F, a^2 + 1);
julia> R, (x, y, z) = polynomial_ring(K, ["x", "y", "z"])
(Singular Polynomial Ring (0,a_1),(x_1,y_1,z_1),(dp(3),C), spoly{n_algExt}[x, y, z])

julia> R.S
3-element Vector{Symbol}:
 :x
 :y
 :z
```
This is in my version that renames all variables, though this occured previously whenever a variable was renamed; we see that the original symbols are properly saved, but the print routine gives us... I'm not even sure what. (But this occurs in Singular.jl, Oscar.jl does not have this problem).